### PR TITLE
Detect host url from conn struct and derive google_redirect_uri issue #17

### PR DESCRIPTION
**Pull request by nelsonic**
Hi @SimonLab as described in #17 the objective of this PR is to:
+ [x] reduce steps in configuring the _consuming_ app (using this package) by detecting the `google_redirect_uri` from the `conn` struct.

### CODE REVIEW

File: lib/elixir_auth_google.ex [hunk 1]
```diff
@@ -8,33 +8,54 @@ defmodule ElixirAuthGoogle do
   @google_token_url "https://oauth2.googleapis.com/token"
   @google_user_profile "https://www.googleapis.com/oauth2/v3/userinfo"
 
+
+  @doc """
+  `get_baseurl_from_conn/1` derives the base URL from the conn struct
+  """
+  @spec get_baseurl_from_conn(Map) :: String.t
+  def get_baseurl_from_conn(conn) do
```
Comment (SimonLab) [inline lib/elixir_auth_google.ex]: I think we could use pattern matching for the parameters and a guard to test the value of the host:
```suggestion
  def get_baseurl_from_conn(%{host: h, port: p}) when host == "localhost" do
    "http://#{h}:#{p}"
  end
  
  def get_baseurl_from_conn(%{host: h}) do
     "https://#{h}"
  end
```

Comment (nelsonic) [reply lib/elixir_auth_google.ex]: That's a much more "elixiry" approach. Thanks! 👍

File: lib/elixir_auth_google.ex [hunk 2]
```diff
@@ -8,33 +8,60 @@ defmodule ElixirAuthGoogle do
   @google_token_url "https://oauth2.googleapis.com/token"
   @google_user_profile "https://www.googleapis.com/oauth2/v3/userinfo"
 
+
+  @doc """
+  `get_baseurl_from_conn/1` derives the base URL from the conn struct
+  """
+  @spec get_baseurl_from_conn(Map) :: String.t
+  def get_baseurl_from_conn(%{host: h, port: p}) when host == "localhost" do
```
Comment (SimonLab) [inline lib/elixir_auth_google.ex]: sorry I did a typo on my first suggestion, it should be:

```suggestion
  def get_baseurl_from_conn(%{host: h, port: p}) when h == "localhost" do
```

Comment (nelsonic) [reply lib/elixir_auth_google.ex]: Yeah, I (_also_) only saw that after I had committed your suggestion ... 🙄 
I've fixed it on `localhost` and thanks to the `pre-commit` hook, I know the tests pass. 😉

**Comment by nelsonic** [comment]
@SimonLab mergeable if you're happy. 👍

**Pull request merged**